aviary/subsystems/propulsion/test/test_propeller_map.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import unittest
+
+from openmdao.components.interp_util.interp_semi import InterpNDSemi
+from openmdao.utils.assert_utils import assert_near_equal
+
+from aviary.subsystems.propulsion.propeller_map import PropellerMap
+from aviary.subsystems.propulsion.utils import PropModelVariables as keys
+from aviary.variable_info.options import get_option_defaults
+from aviary.variable_info.variables import Aircraft
+
+
+class PropellerMapTest(unittest.TestCase):
+    def test_general_aviation(self):
+        tol = 0.005
+        aviary_options = get_option_defaults()
+        prop_file_path = 'models/propellers/general_aviation.prop'
+        aviary_options.set_val(
+            Aircraft.Engine.PROPELLER_DATA_FILE, val=prop_file_path, units='unitless')
+        aviary_options.set_val(
+            Aircraft.Engine.INTERPOLATION_METHOD, val='slinear', units='unitless')
+        aviary_options.set_val(
+            Aircraft.Engine.USE_PROPELLER_MAP, val=True, units='unitless')
+        prop_model = PropellerMap('prop', aviary_options)
+        prop_model.build_propeller_interpolator(3, aviary_options)
+
+        x1 = prop_model.data[keys.MACH]
+        x2 = prop_model.data[keys.CP]
+        x3 = prop_model.data[keys.J]
+        y = prop_model.data[keys.CT]
+        grid = np.array([x1, x2, x3]).T
+        interp = InterpNDSemi(grid, y, method='slinear')
+        # Mach, CP, J from general_aviation, expected CT: 0.0318
+        x = np.array([0.8, 0.025, 0.5])
+        f, df_dx = interp.interpolate(x, compute_derivative=True)
+        assert_near_equal(f, 0.0318, tolerance=tol)
+
+    def test_propfan(self):
+        tol = 0.005
+        aviary_options = get_option_defaults()
+        prop_file_path = 'models/propellers/PropFan.prop'
+        aviary_options.set_val(
+            Aircraft.Engine.PROPELLER_DATA_FILE, val=prop_file_path, units='unitless')
+        aviary_options.set_val(
+            Aircraft.Engine.INTERPOLATION_METHOD, val='slinear', units='unitless')
+        aviary_options.set_val(
+            Aircraft.Engine.USE_PROPELLER_MAP, val=True, units='unitless')
+        prop_model = PropellerMap('prop', aviary_options)
+        prop_model.build_propeller_interpolator(3, aviary_options)
+
+        x1 = prop_model.data[keys.MACH]
+        x2 = prop_model.data[keys.CP]
+        x3 = prop_model.data[keys.J]
+        y = prop_model.data[keys.CT]
+        grid = np.array([x1, x2, x3]).T
+        interp = InterpNDSemi(grid, y, method='slinear')
+        # Mach, CP, J from PropFan, expected CT: 0.095985
+        x = np.array([0.25, 0.0465, 0.2])
+        f, df_dx = interp.interpolate(x, compute_derivative=True)
+        assert_near_equal(f, 0.095985, tolerance=tol)
```

Comment:
We definitely need to include tests of the partial derivatives for all of these new components.

Reply:
@Kenneth-T-Moore , This part of the code is basically `MetaModelSemiStructuredComp` from OpenMDAO. Do we really need to test on partial derivatives? If yes, is there an easy way?

Reply:
You don't need to test the metamodel component, but you do need to test `OutMachs`, and any other components where you define a `compute_partials`.  

Here is a good example of what a derivatives test would look like: https://github.com/OpenMDAO/Aviary/blob/main/aviary/subsystems/aerodynamics/flops_based/test/test_design.py

You will need to do 3 tests, 1 for each different value of "out_type", since each choice has different calculations.

Reply:
I see. Yeah, I added 3 unit tests for OutMachs.